astropy/io/ascii/tests/test_cds_header_from_readme.py

```diff
@@ -21,6 +21,19 @@ def read_table3(readme, data):
     return ascii.read(data, readme=readme)
 
 
+def test_description():
+    readme = 't/cds/description/ReadMe'
+    data = 't/cds/description/table.dat'
+    for read_table in (read_table1, read_table2, read_table3):
+        table = read_table(readme, data)
+        assert_equal(len(table), 10)
+        assert_equal(table['Cluster'].description, 'Cluster name')
+        assert_equal(table['Star'].description, '')
+        assert_equal(table['Wave'].description, 'wave? Wavelength in Angstroms')
+        assert_equal(table['El'].description, 'a')
+        assert_equal(table['ion'].description, '- Ionization stage (1 for neutral element)')
+
```

Comment:
The markup value is processed in line 128 in cds.py.
Currently, the description string is unaltered from what you read in, i.e. the `?=-9.9` is part of the description. What we really should do is something like this:
```
match = re.match(
                     r'\? (?P<equal> =)? (?P<nullval> \S* (?P<descriptiontext> .))', col.description, re.VERBOSE)
col.description = match.group('descriptiontext')
```
(untested, but something like that.)
Not sure how that works out with line breaks in the description 0 I think those are taken care of in an earlier stage. I would just try and see if it works.
Technically is this outside of the scope of your PR ("requiring at least two description characters") and if you don't want to do it we can postpone it to another PR but it would be great if you could take care of it, while you are working on the description stuff anyway.